src/runtime/server/services/utils.ts

```diff
@@ -9,14 +10,17 @@ import { useRuntimeConfig } from '#imports'
  * Get `origin` and fallback to `x-forwarded-host` or `host` headers if not in production.
  */
 export const getServerOrigin = (event?: H3Event): string => {
+  const config = useRuntimeConfig()
+
   // Prio 1: Environment variable
-  const envOrigin = process.env.AUTH_ORIGIN
+  const envOriginKey = camelCase(config.public.auth.originEnvKey!, { normalize: true })
+  const envOrigin = config[envOriginKey] as string|undefined
```

Comment:
It doesn't look correct to me - what would happen when a non-`NUXT_` env variable is used? I would assume that it is not picked up by Nuxt. After consulting with docs, it's exactly the case:

https://nuxt.com/docs/guide/going-further/runtime-config#environment-variables
> Only a specially-named environment variable can override a runtime config property. That is, an uppercase environment variable starting with NUXT_ which uses _ to separate keys and case changes.

I think that you still need to read from `process.env` regardless

```suggestion
  const envOriginKey = config.public.auth.originEnvKey!
  const envOriginKeyCamelcase = camelCase(envOriginKey, { normalize: true })
  const envOrigin: string | undefined = config[envOriginKeyCamelcase] ?? process.env[envOriginKey]
```

Reply:
Yes, if there is no special prefix, the variable will be ignored. It should be so because we work in the Nuxt ecosystem. And it has a special mechanism of working with env variables. And bypassing it by communicating to process seems to me to be a mistake from the architectural point of view. 

Because it may cause side effects. Developers are used to the fact that an application uses variables with a prefix only. And they may not realize that such a variable already exists in the system, it will cause side effects and difficulties.

I am very much against this approach, but if you are ready that side effects appear in the application because of accessing process, then fine.

Reply:
@phoenix-ru leave things as they are? Or accept your changes?

Reply:
Issue with only accepting `NUXT_` prefixes is that some providers (like Vercel/Netlify) set their env variables automatically for preview envs. And it has been quite a requested feature to support them.

By using `process.env` we allow devs to specify `originEnvKey: 'VERCEL_URL'` and NuxtAuth will use this variable.

> Because it may cause side effects

I completely agree with you here and would also love to only use `NUXT_` prefixed env variables and completely ditch `process.env`, but in this case I think we should accommodate for different use-cases. This is also done because `next-auth` itself relies on `process.env`:

https://github.com/nextauthjs/next-auth/blob/064ee18164bbb479bf9727361088f4a8b3738526/packages/core/src/lib/utils/env.ts#L27-L33